test/StateReceiver.test.js

```diff
@@ -0,0 +1,72 @@
+const ethUtils = require('ethereumjs-util')
+const TestStateReceiver = artifacts.require('TestStateReceiver')
+const TestCommitState = artifacts.require('TestCommitState')
+const BN = ethUtils.BN
+
+contract('StateReceiver', async (accounts) => {
```

Comment:
Nice tests. Please try to test a couple more things.
- Calling `commitState` with a non-system address should revert.
- Make `onStateReceive` consume > 5M gas (that's the upper limit on gas we have hardcoded) and assert that `bool success` received from `commitState` is false. Assert true in other scenarios.
- One way to do this could be to decode `_data` in `onStateReceive` and run the loop that many times. To test the above scenario send in a large value so that loop runs out of gas.

Reply:
@atvanguard I've made the changes for the above. Is this what you wanted?